api/offer_test.go

```diff
@@ -385,19 +371,43 @@ var _ = Describe("Offer Handler", func() {
 			Expect(obj["error"]).To(Equal("ValidationFailedError"))
 			Expect(obj["description"]).To(Equal("json: cannot unmarshal string into Go value of type models.OfferToUpdate"))
 		})
+
+		It("should not claim if invalid id is passed on body", func() {
```

Comment:
If the middleware is not passed in app.go (dev mistake), it wouldn't get ID from URL and use maybe invalids IDs from body.